### ISSUE TITLE
spacefm: error while loading shared libraries: libssl.so.47

### ISSUE THREAD

**Issue by madhits** [skycocker/chromebrew#4476]
I searched for answers to the lauching spacefm, and thunar. Nothing applied to chrome OS ver 83 that im running only to arm or other linux distros.

spacefm: error while loading shared libraries: libssl.so.47: cannot open shared object file: No such file or directory

Here is the error I get when I try to lauch thunar.
Thunar
thunar: Failed to initialize Xfconf: Unknown or unsupported transport ?disabled? for address ?disabled:?

## Versions and system information
crew --version
1.4.4

ChromeOS: `cat /etc/lsb-release`
CHROMEOS_RELEASE_APPID={49BA18F3-93DB-4F43-B966-3BBC57881C42}
CHROMEOS_BOARD_APPID={49BA18F3-93DB-4F43-B966-3BBC57881C42}
CHROMEOS_CANARY_APPID={79F28633-123B-4A68-9EC2-F6F56A57A5C5}
DEVICETYPE=CHROMEBOOK
GOOGLE_RELEASE=13020.39.10.38
CHROMEOS_RELEASE_BOARD=amd64-fydeos
CHROMEOS_RELEASE_BRANCH_NUMBER=39
CHROMEOS_RELEASE_TRACK=stable-channel
CHROMEOS_RELEASE_KEYSET=devkeys
CHROMEOS_RELEASE_NAME=Chromium OS
CHROMEOS_AUSERVER=
CHROMEOS_ARC_VERSION=6104653
CHROMEOS_ARC_ANDROID_SDK_VERSION=28
CHROMEOS_DEVSERVER=
CHROMEOS_RELEASE_BUILD_NUMBER=13020
CHROMEOS_RELEASE_CHROME_MILESTONE=83
CHROMEOS_RELEASE_PATCH_NUMBER=10
CHROMEOS_RELEASE_BUILD_TYPE=Release Build v10.2
CHROMEOS_RELEASE_UNIBUILD=1
CHROMEOS_RELEASE_VERSION=13020.39.10.38
CHROMEOS_RELEASE_DESCRIPTION=13020.39.10.38

Architecture: `uname -m`
x86_64

Hope you guys can offer some clues. I have openssl installed so should have libssl so i really dont know why spacefm wont launch. 

Another question do the gui chromebrew apps show up like other apps in chrome os or no?

**Comment by uberhacker** [comment]
@madhits:  We don't support older releases but you could do the following:
```
cd /usr/local/lib64
ln -s libssl.so.48 libssl.so.47
```

**Comment by madhits** [comment]
almost got it.. now it says..
spacefm: error while loading shared libraries: libcrypto.so.45: cannot open shared object file: No such file or directory

so you linked 48 to 47 so can i ln -s 44 to 45? just a guess

**Comment by uberhacker** [comment]
Do you have sommelier configured and running?  Type `crew postinstall sommelier` to find out.

**Comment by supechicken666** [comment]
> almost got it.. I had a few more errors where it was looking for the newer library and i had the old one, i fixed them as you did above via symbolic link.
> 
> once i did the linking i got this error.
> 
> spacefm
> spacefm: Cannot open display
> 
> prob stuck there. my guess is chromebrew wont work for me because its far to dependent on the newer chromeos 85 under pinnings and my only real option for getting working linux apps would be via croton right? thanks for confirming im looking for getting some linux apps on chrome os and hoped chromebrew would do the trick but it seems croton might be it.

Try to add `sommelier --set-x-display=DISPLAY` before `spacefm` command

**Comment by madhits** [comment]
i get 
spacefm: Unknown option --set-x-display=DISPLAY

**Comment by supechicken666** [comment]
> i get
> spacefm: Unknown option --set-x-display=DISPLAY

Sorry, should be `sommelier --x-display=DISPLAY spacefm`

**Comment by madhits** [comment]
> sommelier --x-display=DISPLAY spacefm

It worked, wow. Ok now I see apps run this way can not be reloaded via the gui right? I need to run the command everytime via shell? bummer or is there a workaround for that?

**Comment by uberhacker** [comment]
`sommelier --x-display=$DISPLAY spacefm`?

**Comment by uberhacker** [comment]
If you execute `crew postinstall sommelier` and follow the instructions, sommelier will autostart when you start the shell.

**Comment by madhits** [comment]
> `sommelier --x-display=$DISPLAY spacefm`?

Yes it workedbut I did not have the $ in there only sommelier --x-display=DISPLAY spacefm
 but it appears to work either way. The weird thing is I cant see sda drive, only driveefs and loop1-3 weird. How come I cant see sda1-15

**Comment by madhits** [comment]
> If you execute `crew postinstall sommelier` and follow the instructions, sommelier will autostart when you start the shell.

Ok ill do that but i still cant add spacefm to the gui like any other chrome app can i?

**Comment by uberhacker** [comment]
spacefm should work without prefixing with sommelier just like any other gui app (if you follow the sommelier postinstall instructions).

**Comment by madhits** [comment]
> spacefm should work without prefixing with sommelier just like any other gui app (if you follow the sommelier postinstall instructions).

but do i need to load it from the shell every time or can i pin it to the tray for example? Also i cant get access to the sda15 partitions do I need to make symbolic links into the user space to be able to do that?

**Comment by uberhacker** [comment]
Pinning apps to the tray is one of our Hacktoberfest challenges if you want to give it a shot.  You would be a hero if you can figure it out.  See https://github.com/skycocker/chromebrew/issues/4341.

**Comment by madhits** [comment]
> Pinning apps to the tray is one of our Hacktoberfest challenges if you want to give it a shot. You would be a hero if you can figure it out. See #4341.

ahh ok, makes sense chrome does not even understand the app that is in the tray. So what about seeing partitions outside of the chrome installed partition. Shell can see all of it but it seems spacefm can not. Ive tried linking so files app can see the partition using sympolic links but they never showed up. so is spacefm going to follow that same rule.

**Comment by uberhacker** [comment]
I don't have a good answer to that question.

**Comment by madhits** [comment]
> I don't have a good answer to that question.

ok i might just reinstall then, i currently have the largest partition setup with ubuntu (dual boot) and assumed id be able to see it in chrome but since i cant do it that way if I know ubuntu can see all my chrome partitions easy.

**Comment by satmandu** [comment]
> > If you execute `crew postinstall sommelier` and follow the instructions, sommelier will autostart when you start the shell.
> 
> Ok ill do that but i still cant add spacefm to the gui like any other chrome app can i?
> 
> I was hoping to get access to sda15 its an ext4 partition with 400gb storage.

FYI, as a person with a setup which can dual boot I've noticed that sleep/hibernation does not work if you have a non-ChromeOS partition on the internal drive mounted when the system tries to sleep.

**Comment by satmandu** [comment]
Also, try saving this to /usr/local/bin/libs and run that against your application to tell you what libraries it is using:

```
#!/usr/bin/env bash

needs=`readelf -d $1 2>/dev/null| \
  awk '$2=="(NEEDED)" {
      gsub("[][]", "", $5); 
      hits[$5]=$5
  }
  END {
      for (i in hits) {
          printf(sprintf("%s ", i))
      }
  }'`

/usr/local/sbin/ldconfig -p | \
    awk -vneeds="$needs" \
       'BEGIN{
          split(needs,aneeds)
        }

        { 
          if (!arr[$1]) 
            arr[$1] = $NF
        }

        END { 
          for (i in aneeds) { 
            need = aneeds[i]; 
            if (arr[need]) {
              print need "\t" arr[need]
            } else {
               print "unfulfilled:", need
            }
          }
       }'
```

Here's how it looks on my system:
```
libs `which ssh`
libc.so.6       /usr/local/lib64/libc.so.6
libcrypt.so.1   /usr/local/lib64/libcrypt.so.1
libresolv.so.2  /usr/local/lib64/libresolv.so.2
libcrypto.so.46 /usr/local/lib/libcrypto.so.46
libssp.so.0     /usr/local/lib64/libssp.so.0
libutil.so.1    /usr/local/lib64/libutil.so.1
libdl.so.2      /usr/local/lib64/libdl.so.2
libz.so.1       /usr/local/lib64/libz.so.1
```

**Comment by madhits** [comment]
> 
> 
> > > If you execute `crew postinstall sommelier` and follow the instructions, sommelier will autostart when you start the shell.
> > 
> > 
> > Ok ill do that but i still cant add spacefm to the gui like any other chrome app can i?
> > I was hoping to get access to sda15 its an ext4 partition with 400gb storage.
> 
> FYI, as a person with a setup which can dual boot I've noticed that sleep/hibernation does not work if you have a non-ChromeOS partition on the internal drive mounted when the system tries to sleep. (The system comes back from a crash when you come back from sleep.)

THANK You so much for this info. I actually gave up my dual boot because I needed the access to the files so badly in chrome os. I was switched distros of chrome from fydeos r83 to cloudready r85. Arnold would not boot but since cloudready is r8 I can get full support and dropping the ubuntu is fine because of all the linux options i have in chrome os. So at least i know even if I could mount a hd id lose something like sleep which is very critical.
